registration_api.js

```diff
@@ -0,0 +1,805 @@
+        send({
+            "type": "LOGOUT"
+        });
+    });
+}
+
+const registerLoginButton = document.getElementById('registerLogin');
+if (registerLoginButton) {
+    registerLoginButton.addEventListener('click', () => {
+        redirectToLogin(false, 'EN');
+    });
+};
+
+const getAppList = async () => {
+    const app_id = getSessionAppId();
+    const endpoint = getEndpoint();
+    const api = new DerivAPIBasic({ endpoint, lang: 'EN', app_id });
+    const token1 = getToken();
+
+    // rewrite skeleton to have div inside td with class="skeleton"
+    const skeleton = `<tr>
+        <td><div class="skeleton"></div></td>
+        <td><div class="skeleton"></div></td>
+        <td><div class="skeleton"></div></td>
+        <td><div class="skeleton"></div></td>
+        <td><div class="skeleton"></div></td>
+    </tr>`;
+
+    // create an array with 5 skeleton
+    const skeleton_array = Array(5).fill(skeleton);
+    // for each skeleton create a tr
+    skeleton_array.forEach(item => {
+        const tr = document.createElement('tr');
+        tr.innerHTML = item;
+        document.getElementById('app_list').appendChild(tr);
+    });
+    await api.authorize(token1);
+    const get_data = await api.appList();
+    const app_list = get_data.app_list;
+    // send go to empty state when no app_list
+    if (!app_list.length) {
+        send({
+            "type": "GO_TO_EMPTY_STATE"
+        });
+    };
+    const app_list_body = document.getElementById('app_list');
+    while (app_list_body.firstChild) {
+        app_list_body.removeChild(app_list_body.firstChild);
+    }
+    app_list.forEach((app) => {
+        const { active, app_id, app_markup_percentage, appstore, github, googleplay, homepage, name, redirect_uri, scopes, verification_uri} = app;
+        const tr = document.createElement('tr');
+        tr.innerHTML = `<td>${name}</td>
+                        <td>${app_id}</td>
+                        <td>${scopes.join(', ')}</td>
+                        <td>${redirect_uri}</td>
+                        <td>
+                            <button aria-label="Update app" class="app-btn update-icon" onclick="go_update_mode(${active}, '${app_id}', '${app_markup_percentage}', '${appstore}', '${github}', '${googleplay}', '${homepage}', '${name}', '${redirect_uri}', '${verification_uri}', '${scopes.join(', ')}')"><span>Remove app</span></button>
+                            <button aria-label="Delete app" class="app-btn delete-icon" onclick="open_delete_dialog(${app_id})"><span>Update app</span></button>
+                        </td>
+                        `;
+        app_list_body.appendChild(tr);
+    });
+}
+
+const removeApp = async (app_id) => {
+    const endpoint = getEndpoint();
+    const api = new DerivAPIBasic({ endpoint, lang: 'EN', app_id });
+    const token1 = getToken();
+    await api.authorize(token1);
+    await api.appDelete(app_id);
+    send({ type: 'FETCH_APP_LIST' });
```

Comment:
```suggestion
    send({ type: 'FETCH_APP_LIST' });
    api.disconnect();
```
Since, we are creating connection on each action, we should call the disconnect fn to close the WebSocket connection at the end.